### DISCUSSION TITLE
Disable unity build for specific target

### DISCUSSION THREAD

**Discussion by bruchar1** [mesonbuild/meson#11149] [Q&A] [answered]
Hello!

Is it possible to prevent unity build on a specific target?

For instance, if I have libA, libB and libC, could I build the whole project using unity, but preventing libA to use unity build?

One reason for that is that on a rather large project, I may want to build it using unity, but to rebuild a specific target without unity to ensure I didn't forgot a `#include` somewhere in that library.

Another reason is that I could have a third party subproject that do not work with unity build.

**Comment by eli-schwartz** [comment, marked as answer]
You can use `override_options: ['unity=off']` to force a specific built-in option value on a per-target level.

This is handy for restricting targets that you know don't support unity builds, while leaving all other targets to be set via the global CLI option.
